systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -258,33 +349,43 @@ void testMigrationFromZkToKRaft(ExtensionContext extensionContext) {
+    private void waitForZooKeeperResourcesDeletion(TestStorage testStorage) {
         cmdKubeClient().namespace(testStorage.getNamespaceName())
             .execInCurrentNamespace(Level.INFO, "wait", "--for", "delete", "networkpolicy,serviceaccount,service,secret,configmap,pdb,sps,pvc,pod",
                 "-l", Labels.STRIMZI_NAME_LABEL + "=" + testStorage.getZookeeperStatefulSetName(), "--timeout=300s");
     }
 
-    void assertThatTopicIsPresentInZKMetadata(LabelSelector zkSelector, String topicName) {
-        String zkPodName = kubeClient().listPods(zkSelector).get(0).getMetadata().getName();
-        String commandOutput = cmdKubeClient().execInPod(zkPodName, "./bin/zookeeper-shell.sh", "localhost:12181", "ls", "/brokers/topics").out().trim();
+    private void assertThatTopicIsPresentInZKMetadata(String namespaceName, LabelSelector zkSelector, String topicName) {
+        String zkPodName = kubeClient().namespace(namespaceName).listPods(zkSelector).get(0).getMetadata().getName();
+        String commandOutput = cmdKubeClient().namespace(namespaceName).execInPod(zkPodName, "./bin/zookeeper-shell.sh", "localhost:12181", "ls", "/brokers/topics").out().trim();
 
         assertThat(String.join("KafkaTopic: %s is not present ZK metadata", topicName), commandOutput.contains(topicName), is(true));
     }
 
-    void assertThatTopicIsPresentInKRaftMetadata(LabelSelector controllerSelector, String topicName) {
-        String controllerPodName = kubeClient().listPods(controllerSelector).get(0).getMetadata().getName();
-        String kafkaLogDirName = cmdKubeClient().execInPod(controllerPodName, "/bin/bash", "-c", "ls /var/lib/kafka/data | grep \"kafka-log[0-9]\" -o").out().trim();
-        String commandOutput = cmdKubeClient().execInPod(controllerPodName, "/bin/bash", "-c", "./bin/kafka-dump-log.sh --cluster-metadata-decoder --skip-record-metadata" +
+    private void assertThatTopicIsPresentInKRaftMetadata(String namespaceName, LabelSelector controllerSelector, String topicName) {
+        String controllerPodName = kubeClient().namespace(namespaceName).listPods(controllerSelector).get(0).getMetadata().getName();
+        String kafkaLogDirName = cmdKubeClient().namespace(namespaceName).execInPod(controllerPodName, "/bin/bash", "-c", "ls /var/lib/kafka/data | grep \"kafka-log[0-9]\" -o").out().trim();
+        String commandOutput = cmdKubeClient().namespace(namespaceName).execInPod(controllerPodName, "/bin/bash", "-c", "./bin/kafka-dump-log.sh --cluster-metadata-decoder --skip-record-metadata" +
             " --files /var/lib/kafka/data/" + kafkaLogDirName + "/__cluster_metadata-0/00000000000000000000.log | grep " + topicName).out().trim();
 
         assertThat(String.join("KafkaTopic: %s is not present KRaft metadata", topicName), commandOutput.contains(topicName), is(true));
     }
 
+    private void assertThatClusterMetadataTopicNotPresentInBrokerPod(String namespaceName, LabelSelector brokerSelector) {
```

Comment:
if you get my previous comment, I would make this method more generic to accept a boolean and checking if the metadata topic exist or not depending on that parameter value.